### ISSUE TITLE
Increasing percentage of "Duplicate entry" errors in queued messages

### ISSUE THREAD

**Issue by smclinden** [OpenCTI-Platform/opencti#817]
## Description

The percentage of "Duplicate entry" messages seems to be increasing as the message queue gets smaller. When I started, there were 280,000 queued messages and the rate of processing was about 8 messages/second. Now, after 3 weeks, I am down to 67,000 messages and the rate is between 0 and 0.4 messages/second.

The number of errors in the log files is now greater than the number of successful messages.

## Environment

1. OS: Ubuntu 18.04
2. OpenCTI version: OpenCTI 3.3.2
3. OpenCTI client: frontend
4. Other environment details: Python 3.8

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Run OpenCTS
2. Create a valid MISP connector (this seems to be the source of the duplicates, but I can't be sure)
3. ```
{"type":"WRITE_ERROR","operation_query":"mutation IndicatorAdd($input:IndicatorAddInput){indicatorAdd(input:$input){id stix_id_key entity_type parent_types observableRefs{edges{node{id entity_type stix_id_key 
observable_value}relation{id}}}}}","inner_relation_creation":2,"operation":"IndicatorAdd","time":1732,"variables":{"input":{"name":"accoounts-google.com","description":"","indicator_pattern":"[domain-name:valu
e = 'accoounts-google.com']","main_observable_type":"domain","pattern_type":"stix","valid_from":"2016-06-28T09:28:37Z","valid_until":null,"score":90,"detection":true,"confidence":50,"internal_id_key":null,"sti
x_id_key":"indicator--ca1c137b-39ba-4318-9142-5b47071f3d49","created":"2016-06-28T09:28:37.000Z","modified":"2016-06-28T09:28:37.000Z","createdByRef":"64184d6c-89da-4d33-8ecb-083cd0360f20","markingDefinitions"
:["08ec2ce2-0e19-4ca9-9c8a-841682b1a50c"],"tags":[],"killChainPhases":[]}},"size":587,"error":{"data":{"reason":"Duplicate entry","category":"business","rule":"There is more than one thing of type [Indicator] 
that owns the key [[domain-name:value = 'accoounts-google.com']] of type [indicator_pattern]. "},"stacktrace":["FunctionalError: Business validation","at error (webpack-internal:///./src/config/errors.js:21:10
)","at FunctionalError (webpack-internal:///./src/config/errors.js:53:43)","at eval (webpack-internal:///./src/database/grakn.js:188:83)","at runMicrotasks (<anonymous>)","at processTicksAndRejections (interna
l/process/task_queues.js:97:5)","at async executeWrite (webpack-internal:///./src/database/grakn.js:204:5)","at async createEntity (webpack-internal:///./src/database/grakn.js:2006:25)","at async addIndicator 
(webpack-internal:///./src/domain/indicator.js:195:19)"]},"level":"error","message":"API Call","timestamp":"2020-07-31T11:21:04.754Z"}
```

## Expected Output

I would expect that duplicate messages would be discarded.

## Actual Output

I am uncertain whether erroneous messages are discarded or requeued

**Comment by smclinden** [comment]
Some additional info (6 total workers). There appears to be attempted duplication of effort. Is that expected behavior?

```
worker1.out:INFO:root:Listing Indicators with filters [{"key": "indicator_pattern", "values": ["[domain-name:value = 'accoounts-google.com']"], "operator": "eq"}].
worker1.out:INFO:root:Creating Indicator {accoounts-google.com}.
worker1.out:INFO:root:Listing Indicators with filters [{"key": "indicator_pattern", "values": ["[domain-name:value = 'accoounts-google.com']"], "operator": "eq"}].
worker1.out:INFO:root:Creating Indicator {accoounts-google.com}.
worker3.out:INFO:root:Listing Indicators with filters [{"key": "indicator_pattern", "values": ["[domain-name:value = 'accoounts-google.com']"], "operator": "eq"}].
worker3.out:INFO:root:Creating Indicator {accoounts-google.com}.
worker5.out:INFO:root:Listing Indicators with filters [{"key": "indicator_pattern", "values": ["[domain-name:value = 'accoounts-google.com']"], "operator": "eq"}].
worker5.out:INFO:root:Creating Indicator {accoounts-google.com}.
worker5.out:INFO:root:Listing Indicators with filters [{"key": "indicator_pattern", "values": ["[domain-name:value = 'accoounts-google.com']"], "operator": "eq"}].
worker5.out:INFO:root:Creating Indicator {accoounts-google.com}.

worker2.out:INFO:root:Listing Indicators with filters [{"key": "indicator_pattern", "values": ["[file:hashes.MD5 = '64d55acb693c58656eb3004b595d782c']"], "operator": "eq"}].
worker2.out:INFO:root:Creating Indicator {64d55acb693c58656eb3004b595d782c}.
worker3.out:INFO:root:Listing Indicators with filters [{"key": "indicator_pattern", "values": ["[file:hashes.MD5 = '64d55acb693c58656eb3004b595d782c']"], "operator": "eq"}].
worker3.out:INFO:root:Creating Indicator {64d55acb693c58656eb3004b595d782c}.
worker4.out:INFO:root:Listing Indicators with filters [{"key": "indicator_pattern", "values": ["[file:hashes.MD5 = '64d55acb693c58656eb3004b595d782c']"], "operator": "eq"}].
worker4.out:INFO:root:Creating Indicator {64d55acb693c58656eb3004b595d782c}.
worker6.out:INFO:root:Listing Indicators with filters [{"key": "indicator_pattern", "values": ["[file:hashes.MD5 = '64d55acb693c58656eb3004b595d782c']"], "operator": "eq"}].
worker6.out:INFO:root:Creating Indicator {64d55acb693c58656eb3004b595d782c}.
```

**Comment by SamuelHassine** [comment]
Hello @smclinden,

These errors can be ignored but it is not the expected behaviour. We are working to fix it in the 4.0.

Kind regards,
Samuel

**Comment by smclinden** [comment]
If I understood more of the design, I'd be happy to help. I've been reluctant to delve into it, other than some node passport authentication issues, because I lack a roadmap.
